libimage/push.go

```diff
@@ -50,7 +49,44 @@ func (r *Runtime) Push(ctx context.Context, source, destination string, options
 		destination = resolvedSource
 	}
 
-	logrus.Debugf("Pushing image %s to %s", source, destination)
+	// If specified to push --all-tags, look them up and iterate.
+	if options.AllTags {
+
+		// Do not allow : for tags, other than specifying transport
+		d := strings.TrimPrefix(destination, "docker://")
```

Comment:
This should be done with containers/image functions not via parsing of the image. @vrothberg @mtrmac PTAL